module/data/actor/npc.mjs

```diff
@@ -272,6 +285,9 @@ export default class NPCData extends CreatureTemplate {
     AttributesFields.prepareConcentration.call(this, rollData);
     TraitsFields.prepareResistImmune.call(this);
 
+    // Hit Dice
+    this.attributes.hd.value = Math.max(0, this.attributes.hd.max - this.attributes.hd.spent);
```

Comment:
Could we consider changing this the other way around? So `hd.value` is the field we store, and `hd.spent` is the derived field? It matches our other `value`/`max` fields better, even if it doesn't match the `class` Item's `hitDice`/`hitDiceUsed`.

Reply:
The disadvantage of that approach is we have to make a more deliberate effort when the HP formula or class levels is changed. By storing `spent` then the `value` automatically increases alongside `max`.

Reply:
I see, that sounds like a good improvement then